### PR TITLE
Fix missing ignore textures

### LINKED ISSUE #7304: Ignore item has rendering problems
##### Issue type
 - Bug report
 
##### Minetest version
```
4ccd18d5ec7dec02366a2221526b2fc0ed7b16b6
```

##### OS / Hardware
OS: GNU/Linux
CPU: Intel(R) Core(TM) i7-4510U CPU @ 2.00GHz
GPU model: Haswell-ULT Integrated Graphics Controller
OpenGL version: Mesa 17.3.7

##### Summary
The wielditem image of `ignore` is broken. When you wield `ignore`, you see really strange things in your wield hand, it changes when you walk around. You might see animated water, random node textures, nodes with weird stretching and in generall another mess.

The inventory image is just blank (fully transparent) but should be `ignore.png`.

Also, the tooltop of `ignore` is not matching item's `description`.

In contrast, the `air` item works perfectly fine.

This is a regression from 0.4.16.

##### Steps to reproduce
* `/giveme ignore`
* Look in hotbar at item
* Wield the item, walk around and look at your wieldhand
* Look in inventory at item

### PR DESCRIPTION
**Pull request by HybridDog**
Fixes #7304

### CODE REVIEW

File: src/script/lua_api/l_item.cpp [hunk 1]
```diff
@@ -536,14 +536,18 @@ int ModApiItemMod::l_register_item_raw(lua_State *L)
...
 		ContentFeatures f = read_content_features(L, table);
-		content_t id = ndef->set(f.name, f);
-
-		if(id > MAX_REGISTERED_CONTENT){
-			throw LuaError("Number of registerable nodes ("
-					+ itos(MAX_REGISTERED_CONTENT+1)
-					+ ") exceeded (" + name + ")");
+		// when a mod reregisters ignore, only texture changes and such should
+		// be done
+		if (f.name != "ignore") {
```
Comment (sfan5) [inline src/script/lua_api/l_item.cpp]: <pre>
if (f.name == "ignore")
    return 0;
</pre>

**Comment by HybridDog** [comment]
@sfan5, if I understood you correctly, you want to keep the indentation level low. Please tell me if I'm wrong.
I've put two `return 0;`s (you suggested only one). 
In my opinion, if the indentation levels don't differ significantly, code which is executed more likely (primary order) or code which is usually executed first (secondary order) should be put first, 
in this case the name being `"ignore"` is not very likely, thus the extra return should not be put, and 
the item being a node is about as likely as being not a node and neither a node nor an item which is not a node is usually registered first, so the first return can be put because it reduces the indentation level a bit.

File: src/script/lua_api/l_item.cpp [hunk 2]
```diff
@@ -535,16 +535,22 @@ int ModApiItemMod::l_register_item_raw(lua_State *L)
 	// Register item definition
 	idef->registerItem(def);
 
+	if (def.type != ITEM_NODE)
+		return 0;
```
Comment (sfan5) [inline src/script/lua_api/l_item.cpp]: having the early return here for `ITEM_NODE` implies that a node type for register_item_raw is somehow wrong, I'd keep that here.

Comment (HybridDog) [reply src/script/lua_api/l_item.cpp]: ok

**Pull request merged**